.github/workflows/build-ui.yml

```diff
@@ -0,0 +1,66 @@
+name: Build UI
+
+on:
+  pull_request:
+    branches: [main]
+
```

Comment:
This workflow will not be triggered manually after you apply a particular label to a PR and this condition makes it trigger on every pull request, we don't need that, do we?

Reply:
We can trigger by label as described here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label

Reply:
My thought was that in general, we want this workflow to run everytime, if for nothing else, to do a check for uncommited files.

And in the case where a PR is created with a label (at time of creation, which will be the case for a dependabot PR), we then do some additional work.

That said though, we could potentially put that uncommited file check back in the lint action, and only run this one based on label?

Reply:
I see, that makes sense.